File: Dagger-Android-Sample/src/main/java/com/gabrielsamojlo/dagger_212/di/AppModule.java
package com.gabrielsamojlo.dagger_212.di;

import android.app.Application;
import android.content.Context;

import com.gabrielsamojlo.dagger_212.DummyDependency;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;


@Module
public abstract class AppModule {

    @Binds
    public abstract Context bindContext(Application application);

    @Provides
    static DummyDependency provideDummyDependency(Context context) {
        return new DummyDependency(context);
    }

}

File: Tutorial7-3BindsContextWithApplication/src/main/java/com/smarttoolfactory/tutorial7_3bindscontextwithapplication/di/AppModule.kt
package com.smarttoolfactory.tutorial7_3bindscontextwithapplication.di

import android.app.Application
import android.content.Context
import android.content.SharedPreferences
import dagger.Binds
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module(includes = [AppProviderModule::class])
abstract class AppModule {

    @Binds
    abstract fun bindContext(application: Application): Context

}

@Module
object AppProviderModule {

    @Provides
    @Singleton
    fun provideSharedPreferences(application: Application): SharedPreferences {
        return application.getSharedPreferences("PrefName", Context.MODE_PRIVATE)
    }

}